actor/echo-httpserver-preview2/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+//! Echo Actor component, which compiles to WASI Preview 2
+
+// wit-bindgen generates the traits and implementation that
+// power WIT-based WASM development.
+//
+// To see what wit-bindgen generates, use [cargo expand](https://github.com/dtolnay/cargo-expand)
+wit_bindgen::generate!({
+    world: "echo",
+    exports: {
+        "wasmcloud:bus/guest": EchoActor
+    }
+});
+
+use std::io::{stdin, stdout, Write};
+
+use serde_json::json;
+use wasmcloud_actor::{debug, error};
+use wasmcloud_interface_httpserver::{HttpRequest, HttpResponse};
+
+/// Operation used by wasmCloud httpserver capability contract
+/// for handling a single request
+/// see [wasmCloud/interfaces httpserver](https://github.com/wasmCloud/interfaces/tree/main/httpserver#-rust)
+const OPERATION_HTTPSERVER_HANDLE_REQUEST: &str = "HttpServer.HandleRequest";
+
+/// Echo (HTTP) actor
+///
+/// This unit struct will hold all implementations of
+/// traits generated the by WIT contract (see echo-wasi-preview2/wit/echo.wit)
+struct EchoActor;
+
+/// Implementation of a wasmCloud guest module (guest module) that can
+/// receive messages (RPCs) from a wasmCloud lattice.
+///
+/// NOTE: this implementation follows closely the tests written
+/// for wasmcloud/wasmcloud, specific to reactor components.
+/// https://github.com/wasmCloud/wasmCloud/blob/main/tests/actors/rust/builtins-component-reactor/src/lib.rs
+impl exports::wasmcloud::bus::guest::Guest for EchoActor {
```

Comment:
I would expect this to export `wasi:http/incoming-handler`, users should not manually implement `wasmcloud:bus/guest`

Reply:
Sure that's reasonable -- now that `wasi:http/incoming-handler` exists it's reasonable to use that instead.

- [ ] switch to using `wasi:http/incoming-handler`